src/tools/tidy/src/error_codes_check.rs

```diff
@@ -90,14 +98,26 @@ fn extract_error_codes(
                 .split_once(':')
                 .expect(
                     format!(
-                        "Expected a line with the format `E0xxx: include_str!(\"..\")`, but got {} without a `:` delimiter",
+                        "Expected a line with the format `E0xxx: include_str!(\"..\")`, but got {} \
+                         without a `:` delimiter",
                         s,
-                    ).as_str()
+                    )
+                    .as_str(),
                 )
                 .0
                 .to_owned();
-            if !error_codes.contains_key(&err_code) {
-                error_codes.insert(err_code.clone(), false);
+            match error_codes.entry(err_code.clone()) {
+                Entry::Occupied(mut e) => {
+                    let mut entry = e.get_mut();
+                    entry.has_explanation = true
+                }
+                Entry::Vacant(e) => {
+                    e.insert(ErrorCodeStatus {
+                        has_test: false,
+                        is_used: false,
+                        has_explanation: true,
+                    });
+                }
```

Comment:
Can we rewrite this logic to do something like `error_codes.entry(err_code.clone()).or_default().has_explanation = true;`? That would give us the same result I believe in a more readable (IMO) fashion.